lib/ASRouter.jsm

```diff
@@ -696,11 +760,8 @@ class _ASRouter {
             storage: this._storage,
             dispatchToAS: this.dispatchToAS,
           });
-          messages = messages.filter(
-            ({ content }) =>
-              !content ||
-              !content.category ||
-              ASRouterPreferences.getUserPreference(content.category)
+          messages = messages.filter(({ groups }) =>
+            this.hasGroupsEnabled(groups)
```

Comment:
The consequence of this is that a message that is part of a disabled group will be filtered out. This might be confusing when using ASRAdmin to debug messages.

Reply:
Don't filter out here, do it in getUnblockedMessages so we keep the messages around in ASRAdmin and we can explain the state.

Reply:
Move the `exclude` provider configuration in there as well.